pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
-
-    def select_accelerator(self) -> Accelerator:
-        if isinstance(self.distributed_backend, Accelerator):
-            # custom accelerator from user
-            if self._precision_plugin is not None or self._strategy is not None:
-                # plugins also specified by user
-                rank_zero_warn(
-                    "Specified `Precision` and `TrainingType` plugins will be ignored,"
-                    " since an `Accelerator` instance was provided."
+        if (
+            self._precision_flag == 16
+            and isinstance(self.accelerator, CPUAccelerator)
+            and self._amp_type_flag == AMPType.APEX
+        ):
+            raise MisconfigurationException(
+                "You passed `Trainer(accelerator='cpu', precision=16, amp_type='apex')`"
+                " but apex AMP not supported on CPU."
+            )
+        if self._precision_flag == "bf16" and self._amp_type_flag != AMPType.NATIVE:
+            raise MisconfigurationException(
+                f"You passed `Trainer(amp_type={self._amp_type_flag.value!r}, precision='bf16')` but "
+                "it's not supported. Try using `amp_type='native'` instead."
+            )
+        if self._precision_flag in (16, "bf16") and self._amp_type_flag == AMPType.APEX:
+            if isinstance(self.strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy, DDPFullyShardedStrategy)):
+                raise MisconfigurationException(
+                    "Sharded plugins are not supported with apex, please switch to `amp_backend='native'`."
                 )
-            return self.distributed_backend
-
-        if self.use_gpu:
-            acc_cls = GPUAccelerator
-        elif self.use_tpu:
-            acc_cls = TPUAccelerator
-        elif self.use_ipu:
-            acc_cls = IPUAccelerator
-        else:
-            acc_cls = CPUAccelerator
-
-        accelerator = acc_cls()
-        return accelerator
-
-    def select_cluster_environment(self) -> ClusterEnvironment:
-        if self._cluster_environment is not None:
-            return self._cluster_environment
-        if self._is_slurm_managing_tasks():
-            rank_zero_info("Multiprocessing is handled by SLURM.")
-            return SLURMEnvironment()
 
-        for env_type in (BaguaEnvironment, TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment):
-            if env_type.detect():
-                return env_type()
-
-        return LightningEnvironment()
+    def _lazy_init_strategy(self) -> None:
+        """Lazily set missing attributes on the previously instantiated strategy."""
+        self.strategy.accelerator = self.accelerator
+        if self.precision_plugin:
+            self.strategy.precision_plugin = self.precision_plugin
+        if self.checkpoint_io:
+            self.strategy.checkpoint_io = self.checkpoint_io
+        if hasattr(self.strategy, "cluster_environment"):
+            self.strategy.cluster_environment = self.cluster_environment
```

Comment:
noob question. If the strategy doesn't take a cluster_environement, none should be provided right ?